nifi-extension-bundles/nifi-snowflake-bundle/nifi-snowflake-services/src/main/java/org/apache/nifi/snowflake/service/SnowflakeComputingConnectionPool.java

```diff
@@ -190,6 +204,18 @@ protected Collection<ValidationResult> customValidate(final ValidationContext co
         return Collections.emptyList();
     }
 
+    @OnEnabled
+    public void onConfigured(final ConfigurationContext context) throws InitializationException {
+        super.onConfigured(context);
+        accessTokenProvider = context.getProperty(ACCESS_TOKEN_PROVIDER).asControllerService(OAuth2AccessTokenProvider.class);
+    }
```

Comment:
I suggest setting `accessTokenProvider` reference to `null` in `@OnDisabled` method. Even though it does not cause any issues in the code logic, it is good practice to clean up and not to hold references while the controller service is disabled.